src/arrayMethodJoin.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
-/**
- * Implement method join
- */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null) {
+        if (this[i] !== undefined) {
+          result += this[i];
+        }
+      }
+
+      if (i !== this.length - 1) {
+        if (separator !== undefined) {
+          result += separator;
+        }
+      }
```

Comment:
```suggestion
      if (i !== this.length - 1 && separator !== undefined) {
        result += separator;
      }
```
avoid nesting, use logical operators